### DISCUSSION TITLE
includeFiles in build vercel.json

### DISCUSSION THREAD

**Discussion by priyang12** [vercel/vercel#9069] [Help] [answered]
i want `includeFiles` in builds property in vercel.json. i am trying deploy a node function which uses the google API key but I need the serverless function to access keys.json. i saw on the docs that I can add files to output by `includeFiles` but it's not available In  "builds" and if I use it `functions` using it started looking for public dir. how to i point it to use the index.js which a node function?

**Comment by TooTallNate** [comment, marked as answer]
For your case you shouldn't need `includeFiles`, since the bundler will automatically notice that you are using that file and bundle it into the Serverless Function implicitly. Be sure to use `fs.readFileSync()` or similar to trigger the bundling mechanism, something like:

```js
import fs from 'fs';

const keys = JSON.parse(fs.readFileSync(__dirname + '/keys.json'));

export default (req, res) => { … }
```

To answer your original question though, `includeFiles` property belongs in the `config` object with the `builds` array. So something like:

```json
{
  "builds": [
    {
      "src": "api/*.js",
      "use": "@vercel/node",
      "config": {
        "includeFiles": [ "keys.json" ]
      }
    }
  ]
}
```

Note that you can also achieve similar with the [`"functions"` config](https://vercel.com/docs/project-configuration#project-configuration/functions) within `vercel.json`, which is the preferred syntax over the legacy `builds` property.

Hope that helps. Cheers!
